### PR TITLE
Added basic link duplication prevention

### PR DESCRIPTION
**Pull request by Timer00**
This can already be merged I believe, still I am looking for ways of making the duplicate url detection more flexible.

### CODE REVIEW

**Comment by Timer00** [comment]
I was unsure about where to place the helper functions I wrote for the flexible URL detection. I see there is a /utils/ folder but none of the files there seemed to match this scenario. Also, if I were to place them there I'd probably need to make the functions more general?

File: components/Forms/validators.ts [hunk 1]
```diff
@@ -47,6 +50,32 @@ export const usernameTaken = async (username: string, wallet: string) => {
...
+    if (array[i] === el) return true;
+  }
+  return false;
+}
+
+export const urlDuplicate = async (mediaURI) => {
+  const query = (await supabase.from("submissions").select("mediaURI")).data.map(d=> d.mediaURI);
+  return isInArray(query.map(cleanUrl), cleanUrl(mediaURI.href));
+};
+
```
Comment (Andrewjeska) [inline components/Forms/validators.ts:78]: Like the changes here! But you can just use `.includes`, no? 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

**Review by Andrewjeska — APPROVED** [review]
can merge after you use .includes or .indexOf or something

File: components/Forms/validators.ts [hunk 2]
```diff
@@ -18,6 +18,9 @@ export const validateSubmission = (values: Submission) => {
     if (!!url && url?.protocol !== "http:" && url?.protocol !== "https:") {
       errors.mediaURI = "Only http or https";
     }
+    if (await urlDuplicate(url)){
+      errors.mediaURI = "Duplicate submission";
+    }
```
Comment (Andrewjeska) [inline components/Forms/validators.ts:23]: We should only run this check if `errors.mediaURI` isn't set already, that way we avoid making the call to supabase.

File: components/Forms/validators.ts (hunk 1 above)
Comment (Andrewjeska) [inline components/Forms/validators.ts:78]: Okay so here's the problem, you're getting all submissions, you need to learn about `.match`
```
await supabase.from("submissions").select().match({mediaURI: cleanUrl(mediaURI.href});
```

**Review by Andrewjeska — APPROVED** [review]

**Pull request merged**